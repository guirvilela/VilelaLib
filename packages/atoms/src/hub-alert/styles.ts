import styled from 'styled-components';

import { HubAlertProps } from './types';

export const AlertInnerStyle = styled.div`
  display: flex;
  gap: ${({ theme }) => theme.spaces[2]};
  margin-top: -2px;
  width: 100%;
`;

export const AlertTitleStyle = styled.span`
  font-family: ${({ theme }) => theme.typography.body2.family};
  font-size: ${({ theme }) => theme.typography.body2.size};
  line-height: ${({ theme }) => theme.typography.body2.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body2.letterSpacing};
  color: ${({ theme }) => theme.colors.neutral[800]};
`;

export const AlertContentStyle = styled.span`
  font-family: ${({ theme }) => theme.typography.body2.family};
  font-size: ${({ theme }) => theme.typography.body2.size};
  line-height: ${({ theme }) => theme.typography.body2.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body2.letterSpacing};
  color: ${({ theme }) => theme.colors.neutralGrey[700]};
  font-weight: 400;
  width: 100%;
  text-overflow: ellipsis;
  overflow: hidden;
`;

export const AlertCloseStyle = styled.button`
  background-color: transparent;
  width: ${({ theme }) => theme.spaces[3]};
  height: ${({ theme }) => theme.spaces[3]};
  border: none;
  color: inherit;
  cursor: pointer;
`;

export const AlertStyle = styled.div<Omit<HubAlertProps, 'title'>>`
  display: flex;
  width: 100%;
  background-color: ${({ theme, color }) =>
    color ? theme.colors[color][100] : theme.colors.neutral[100]};
  color: ${({ theme, color }) =>
    color ? theme.colors[color][500] : theme.colors.neutral[500]};
  border: 1px solid
    ${({ theme, color }) =>
      color ? theme.colors[color][500] : theme.colors.neutral[500]};
  border-radius: ${({ theme, radius }) =>
    radius ? theme.spaces[radius] : '4px'};
  align-items: ${({ inline }) => (inline ? 'center' : 'start')};

  ${({ theme, pad }) => {
    if (typeof pad === 'string') {
      return `padding: ${theme.spaces[pad]};`;
    }
    if (Array.isArray(pad)) {
      return `padding: ${pad.reduce(
        (acc, x) => acc + `${theme.spaces[x]} `,
        ''
      )};`;
    }
    return `padding: ${theme.spaces[2]};`;
  }}

  gap: ${({ theme, gap }) => (gap ? theme.spaces[gap] : theme.spaces[2])};

  ${AlertInnerStyle} {
    flex-direction: ${({ inline }) => (inline ? 'row' : 'column')};
    align-items: ${({ inline }) => (inline ? 'center' : 'start')};
  }

  ${AlertTitleStyle} {
    font-weight: ${({ inline }) => (inline ? 400 : 600)};
    white-space: ${({ nowrap }) => (nowrap ? 'nowrap' : undefined)};
  }
`;
