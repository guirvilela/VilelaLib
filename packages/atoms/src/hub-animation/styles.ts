import styled from 'styled-components';

import { HubAnimationProps } from './types';

export const AnimationStyle = styled.div<HubAnimationProps & { ref }>`
  ${({ theme, size, width, height }) => {
    switch (size) {
      case 'xs':
        return `
          width: ${theme.spaces[2]};
          height: ${theme.spaces[2]};
          min-width: ${theme.spaces[2]};
        `;
      case 'sm':
        return `
          width: ${theme.spaces[3]};
          height: ${theme.spaces[3]};
          min-width: ${theme.spaces[3]};
        `;
      case 'md':
        return `
          width: ${theme.spaces[5]};
          height: ${theme.spaces[5]};
          min-width: ${theme.spaces[5]};
        `;
      case 'lg':
        return `
          width: ${theme.spaces[6]};
          height: ${theme.spaces[6]};
          min-width: ${theme.spaces[6]};
        `;
      default:
        return `
          width: ${width || theme.spaces[5]};
          height: ${height || theme.spaces[5]};
          min-width: ${width || theme.spaces[5]};
        `;
    }
  }}
`;
