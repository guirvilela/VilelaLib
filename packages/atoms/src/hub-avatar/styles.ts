import styled from 'styled-components';

import { HubAvatarProps } from './types';

export const AvatarImgStyle = styled.img`
  width: 100%;
  height: 100%;
  background-size: cover;
  border-radius: 50%;
`;

export const AvatarInitialsStyle = styled.span`
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  border-radius: 50%;
`;

export const AvatarStyle = styled.div<HubAvatarProps>`
  ${({ theme, size }) => {
    if (!size) {
      return `
        width: ${theme.spaces[7]};
        height: ${theme.spaces[7]};
        min-width: ${theme.spaces[7]};
      `;
    }

    switch (size) {
      case 'xs':
        return `
          width: ${theme.spaces[3]};
          height: ${theme.spaces[3]};
          min-width: ${theme.spaces[3]};
        `;
      case 'sm':
        return `
          width: ${theme.spaces[5]};
          height: ${theme.spaces[5]};
          min-width: ${theme.spaces[5]};
        `;
      case 'md':
        return `
          width: ${theme.spaces[7]};
          height: ${theme.spaces[7]};
          min-width: ${theme.spaces[7]};
        `;
      case 'lg':
        return `
          width: ${theme.spaces[10]};
          height: ${theme.spaces[10]};
          min-width: ${theme.spaces[10]};
        `;
      default:
        return `
          width: ${size};
          height: ${size};
          min-width: ${size};
        `;
    }
  }}
  ${({ borderless, theme }) =>
    borderless
      ? ''
      : `
    padding: ${theme.spaces[1]};
    border: 1px solid ${theme.colors.neutral[400]};
  `};

  border-radius: 50%;

  ${AvatarInitialsStyle} {
    ${({ theme, typo }) => `
      font-family: ${theme.typography[typo || 'h4'].family};
      font-size: ${theme.typography[typo || 'h4'].size};
      line-height: ${theme.typography[typo || 'h4'].lineHeight};
      letter-spacing: ${theme.typography[typo || 'h4'].letterSpacing};
    `}

    background-color: ${({ theme, color, colorLevel }) =>
      color && colorLevel
        ? theme.colors[color][colorLevel]
        : color && theme.colors[color]
        ? theme.colors[color][500]
        : color
        ? color
        : undefined};

    color: ${({ theme, colorText, colorTextLevel }) =>
      colorText && colorTextLevel
        ? theme.colors[colorText][colorTextLevel]
        : colorText && theme.colors[colorText]
        ? theme.colors[colorText][500]
        : colorText
        ? colorText
        : theme.colors.neutral[800]};
  }
`;
