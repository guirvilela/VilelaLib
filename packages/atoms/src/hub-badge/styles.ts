import styled, { css } from 'styled-components';

import { HubBadgeProps } from './types';

const allStyle = css<HubBadgeProps>`
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  gap: ${({ theme }) => theme.spaces[2]};

  ${({ theme, size, width, height }) => {
    if (width || height) {
      return `
        width: ${width};
        height: ${height};
      `;
    }

    switch (size) {
      case 'xs':
        return `
          width: ${theme.spaces[2]};
          height: ${theme.spaces[2]};
        `;
      case 'sm':
        return `
          width: ${theme.spaces[3]};
          height: ${theme.spaces[3]};
        `;
      case 'md':
        return `
          width: ${theme.spaces[4]};
          height: ${theme.spaces[4]};
        `;
      case 'lg':
        return `
          width: ${theme.spaces[5]};
          height: ${theme.spaces[5]};
        `;
      default:
        return '';
    }
  }}

  background-color: ${({ theme, color, colorLevel }) =>
    color && colorLevel
      ? theme.colors[color][colorLevel]
      : color
      ? theme.colors[color][100]
      : theme.colors.neutral[200]};

  color: ${({ theme, color, textColor, textColorLevel }) =>
    textColor && textColorLevel
      ? theme.colors[textColor][textColorLevel]
      : textColor
      ? theme.colors[textColor][500]
      : color
      ? theme.colors[color][500]
      : theme.colors.neutral[500]};

  border-radius: ${({ theme, rounded, radius }) =>
    rounded ? '50%' : theme.spaces[radius || '1']};

  ${({ theme, pad }) => {
    if (!pad) return `padding: 2px 6px`;
    if (typeof pad === 'string') return `padding: ${theme.spaces[pad]}`;
    if (pad.length === 4) {
      return `padding: ${theme.spaces[pad[0]]} ${theme.spaces[pad[1]]} ${
        theme.spaces[pad[2]]
      } ${theme.spaces[pad[3]]}`;
    }
    if (pad.length === 3) {
      return `padding: ${theme.spaces[pad[0]]} ${theme.spaces[pad[1]]} ${
        theme.spaces[pad[2]]
      }`;
    }
    if (pad.length === 2) {
      return `padding: ${theme.spaces[pad[0]]} ${theme.spaces[pad[1]]}`;
    }
    return `padding: ${theme.spaces[pad[0]]}`;
  }};

  font-family: ${({ theme }) => theme.typography.body2.family};
  font-size: ${({ theme }) => theme.typography.body2.size};
  line-height: ${({ theme }) => theme.typography.body2.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body2.letterSpacing};
  font-weight: 600;
`;

const floatStyle = css<HubBadgeProps>`
  position: relative;

  &::after {
    ${allStyle}

    position: absolute;
    content: '${({ content }) => content}';
    border: 1px solid ${({ theme }) => theme.colors.white[500]};
    ${({ theme, position }) => {
      if (position) {
        if (Array.isArray(position)) {
          return `
            top: -${theme.spaces[position[0]]};
            right: -${theme.spaces[position[1]]};
          `;
        }
        return `
          top: -${theme.spaces[position]};
          right: -${theme.spaces[position]};
        `;
      }
      return `
        top: -10px;
        right: -10px;
      `;
    }}
    ${({ theme }) => `
      min-width: ${theme.spaces[4]};
      width: ${theme.spaces[4]};
      height: ${theme.spaces[4]};
    `}
    padding: 0;

    font-family: ${({ theme }) => theme.typography.body3.family};
    font-size: ${({ theme }) => theme.typography.body3.size};
    line-height: ${({ theme }) => theme.typography.body3.lineHeight};
    letter-spacing: ${({ theme }) => theme.typography.body3.letterSpacing};
  }
`;

export const BadgeStyle = styled.div<HubBadgeProps>`
  ${({ float, content }) => {
    if (float) {
      if (content) {
        return floatStyle;
      }
    } else {
      return allStyle;
    }
    return '';
  }}
`;
