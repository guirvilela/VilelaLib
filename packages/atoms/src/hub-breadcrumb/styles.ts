import styled from 'styled-components';

import { HubBreadcrumbProps, HubBreadcrumbsProps } from './types';

export const BreadcrumbsStyle = styled.div<HubBreadcrumbsProps>`
  display: flex;
  align-items: center;
  overflow: auto;
`;

export const BreadcrumbsPathStyle = styled.div`
  display: flex;
  align-items: center;
`;

export const BreadcrumbsIconStyle = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  color: ${({ theme }) => theme.colors.neutral[400]};
  margin: ${({ theme }) => `0 ${theme.spaces[2]}`};
`;

export const BreadcrumbStyle = styled.span<HubBreadcrumbProps>`
  display: flex;
  align-items: center;
  gap: ${({ theme }) => theme.spaces[2]};
  font-family: ${({ theme }) => theme.typography.body2.family};
  font-size: ${({ theme }) => theme.typography.body2.size};
  line-height: ${({ theme }) => theme.typography.body2.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body2.letterSpacing};
  color: ${({ theme, active }) =>
    active ? theme.colors.primary[500] : theme.colors.neutral[400]};
  transition: color 0.5s;
  font-weight: 600;
  white-space: ${({ nowrap }) => (nowrap ? 'nowrap' : 'pre-wrap')};

  ${({ width }) =>
    width
      ? `
    display: block;
    max-width: ${width};
    text-overflow: ellipsis;
    overflow: hidden;
    white-space: nowrap;
  `
      : ''}

  &:hover {
    color: ${({ theme }) => theme.colors.primary[500]};
  }
`;
