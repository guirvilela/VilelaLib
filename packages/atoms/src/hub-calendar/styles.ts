import styled from 'styled-components';

import { HubCalendarItem, HubCalendarProps } from './types';

export const CalendarHeadStyle = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: ${({ theme }) => theme.spaces[3]};
  width: 100%;
`;

export const CalendarHeadTitleStyle = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  gap: ${({ theme }) => theme.spaces[2]};

  font-family: ${({ theme }) => theme.typography.h4.family};
  font-size: ${({ theme }) => theme.typography.h4.size};
  line-height: ${({ theme }) => theme.typography.h4.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.h4.letterSpacing};
  font-weight: 500;
`;

export const CalendarHeadLinkStyle = styled.button`
  background-color: transparent;
  color: ${({ theme }) => theme.colors.primary[200]};
  border: none;

  :hover {
    text-decoration: underline;
    cursor: pointer;
  }
`;

export const CalendarHeadNextStyle = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;
  border: none;
  background-color: ${({ theme }) => theme.colors.neutral[200]};
  border-radius: 50%;
  cursor: pointer;

  height: ${({ theme }) => theme.spaces[6]};
  width: ${({ theme }) => theme.spaces[6]};
`;

export const CalendarHeadPrevStyle = styled(CalendarHeadNextStyle)`
  transform: rotate(-180deg);
`;

export const CalendarDaysStyle = styled.div`
  display: grid;
  align-items: center;
  justify-content: center;
  justify-items: center;
  gap: ${({ theme }) => theme.spaces[2]};
  grid-template-columns: auto auto auto auto auto auto auto;
  width: 100%;
`;

export const CalendarMonthsStyle = styled.div`
  display: grid;
  grid-template-columns: 50% 50%;
  width: 100%;
`;

export const CalendarYearsStyle = styled.div`
  display: grid;
  grid-template-columns: 50% 50%;
  width: 100%;
`;

export const CalendarPeriodStyle = styled.span`
  display: flex;
  align-items: center;
  justify-content: center;

  font-family: ${({ theme }) => theme.typography.body3.family};
  font-size: ${({ theme }) => theme.typography.body3.size};
  line-height: ${({ theme }) => theme.typography.body3.lineHeight};
  letter-spacing: ${({ theme }) => theme.typography.body3.letterSpacing};
  font-weight: 400;
`;

export const CalendarDayStyle = styled.button<HubCalendarItem>`
  display: flex;
  align-items: center;
  justify-content: center;

  color: ${({
    theme,
    selected,
    active,
    previous,
    next,
    disabled,
    today,
    event
  }) => {
    if (selected || active) {
      if (previous || next) {
        if (event && event.color) {
          return theme.colors[event.color][300];
        }
        return theme.colors.neutral[500];
      }
      return theme.colors.white[500];
    }
    if (previous || next) {
      if (event && event.color) {
        return theme.colors[event.color][200];
      }
      return theme.colors.neutral[300];
    }
    if (disabled) {
      if (event && event.color) {
        return theme.colors[event.color][400];
      }
      return theme.colors.neutral[300];
    }
    if (today || (event && event.color)) {
      if (event && event.color) {
        return theme.colors[event.color][500];
      }
      return theme.colors.info[600];
    }
    return theme.colors.neutral[800];
  }};
  background-color: ${({
    theme,
    selected,
    active,
    today,
    event,
    previous,
    next
  }) => {
    if (previous || next) {
      if (event && event.color) {
        return theme.colors[event.color][400];
      }
    }
    if (selected) {
      if (event && event.color) {
        return theme.colors[event.color][500];
      }
      return theme.colors.info[500];
    }
    if (active) {
      if (event && event.color) {
        return theme.colors[event.color][300];
      }
      return theme.colors.info[300];
    }
    if (today || (event && event.color)) {
      if (event && event.color) {
        return theme.colors[event.color][100];
      }
      return theme.colors.info[100];
    }
    return 'transparent';
  }};
  border-radius: 50%;
  border: none;
  ${({ disabled }) => (disabled ? 'cursor: not-allowed;' : `cursor: pointer;`)};
  transition: 0.5s ease background-color, 0.2s ease color;
`;

export const CalendarMonthYearStyle = styled.button<HubCalendarItem>`
  display: flex;
  align-items: center;
  justify-content: center;
  padding: ${({ theme }) => `${theme.spaces[1]} ${theme.spaces[2]}`};
  background-color: transparent;
  color: ${({ theme, selected, active }) => {
    if (selected) {
      return theme.colors.info[500];
    }
    if (!active) {
      return theme.colors.neutral[300];
    }
    return theme.colors.neutral[800];
  }};
  border-radius: 50%;
  border: none;
  cursor: pointer;

  :hover {
    text-decoration: underline;
  }
`;

export const CalendarActionsStyle = styled.div`
  display: grid;
  align-items: center;
  justify-content: end;
  gap: ${({ theme }) => theme.spaces[2]};
  width: 100%;
  padding: ${({ theme }) => `0 ${theme.spaces[2]}`};
`;

export const CalendarTodayStyle = styled.button`
  border: none;
  background-color: transparent;
  color: ${({ theme, disabled }) => {
    if (disabled) {
      return theme.colors.neutral[300];
    }
    return theme.colors.primary[200];
  }};
  ${({ disabled }) => (disabled ? '' : `cursor: pointer;`)};
`;

export const CalendarStyle = styled.div<HubCalendarProps>`
  display: flex;
  align-items: center;
  flex-direction: column;
  gap: ${({ theme }) => theme.spaces[2]};
  width: ${({ width }) => width || '100%'};

  ${CalendarDayStyle} {
    ${({ theme, size }) => {
      switch (size) {
        case 'xs':
          return `width: ${theme.spaces[2]}; height: ${theme.spaces[2]};`;
        case 'sm':
          return `width: ${theme.spaces[3]}; height: ${theme.spaces[3]};`;
        case 'md':
          return `width: ${theme.spaces[5]}; height: ${theme.spaces[5]};`;
        case 'lg':
          return `width: ${theme.spaces[6]}; height: ${theme.spaces[6]};`;
        default:
          if (size) {
            return `width: ${size}; height: ${size};`;
          }
          return `width: ${theme.spaces[6]}; height: ${theme.spaces[6]};`;
      }
    }};
  }

  ${CalendarTodayStyle},
  ${CalendarDayStyle} {
    font-family: ${({ theme, typo }) => theme.typography[typo || 'h4'].family};
    font-size: ${({ theme, typo }) => theme.typography[typo || 'h4'].size};
    line-height: ${({ theme, typo }) =>
      theme.typography[typo || 'h4'].lineHeight};
    letter-spacing: ${({ theme, typo }) =>
      theme.typography[typo || 'h4'].letterSpacing};
    font-weight: ${({ typo }) =>
      typo ? (typo.includes('body') ? 600 : 400) : 400};
  }

  ${CalendarHeadTitleStyle},
  ${CalendarHeadLinkStyle} {
    font-family: ${({ theme, typo, typoHeader }) =>
      theme.typography[typoHeader || typo || 'h4'].family};
    font-size: ${({ theme, typo, typoHeader }) =>
      theme.typography[typoHeader || typo || 'h4'].size};
    line-height: ${({ theme, typo, typoHeader }) =>
      theme.typography[typoHeader || typo || 'h4'].lineHeight};
    letter-spacing: ${({ theme, typo, typoHeader }) =>
      theme.typography[typoHeader || typo || 'h4'].letterSpacing};
    font-weight: ${({ typo, typoHeader }) =>
      typoHeader || typo
        ? (typoHeader || typo)?.includes('body')
          ? 600
          : 400
        : 400};
  }
`;
