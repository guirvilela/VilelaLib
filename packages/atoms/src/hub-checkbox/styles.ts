import styled from 'styled-components';

import { HubCheckboxProps } from './types';

export const CheckboxIconStyle = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  transition: background-color 0.5s;
`;

export const CheckboxInputStyle = styled.input`
  display: none;
`;

export const CheckboxLabelStyle = styled.label`
  font-family: ${(props) => props.theme.typography.body2.family};
  font-size: ${(props) => props.theme.typography.body2.size};
  line-height: ${(props) => props.theme.typography.body2.lineHeight};
  letter-spacing: ${(props) => props.theme.typography.body2.letterSpacing};
  font-weight: 400;
  margin-left: ${(props) => props.theme.spaces[2]};
`;

export const CheckboxStyle = styled.div<
  HubCheckboxProps & { clickable: boolean }
>`
  display: flex;
  align-items: center;
  justify-content: ${({ align }) => align || 'start'};
  ${({ clickable, disabled }) =>
    disabled
      ? 'cursor: not-allowed;'
      : clickable
      ? 'cursor: pointer;'
      : undefined}

  ${CheckboxIconStyle} {
    background-color: ${(props) => {
      if (!props.checked && props.disabled) {
        return props.color
          ? props.theme.colors[props.color][100]
          : props.theme.colors.neutral[200];
      }
      if (props.checked && props.disabled) {
        return props.theme.colors[props.color || 'neutral'][400];
      }
      if (props.checked) {
        return props.theme.colors[props.color || 'primary'][500];
      }
      return props.theme.colors.white[500];
    }};
    color: ${(props) => props.theme.colors[props.colorText || 'white'][500]};
    border: 1px solid
      ${(props) => {
        if (!props.checked || props.disabled) {
          return props.theme.colors[
            props.borderColor || props.color || 'neutral'
          ][400];
        }
        return props.theme.colors[props.borderColor || 'transparent'][500];
      }};
    border-radius: 4px;
    ${(props) => `
      min-width: ${
        props.size === 'lg' ? props.theme.spaces[5] : props.theme.spaces[3]
      };
      width: ${
        props.size === 'lg' ? props.theme.spaces[5] : props.theme.spaces[3]
      };
      height: ${
        props.size === 'lg' ? props.theme.spaces[5] : props.theme.spaces[3]
      };
    `}
  }
`;
