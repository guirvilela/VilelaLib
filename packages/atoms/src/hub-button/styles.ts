import { shadeColor } from '@hubert/shared';
import { animations } from '@hubert/styles';
import styled, { css } from 'styled-components';

import { HubButtonProps } from './types';

const ButtonSizesStyle = css<HubButtonProps>`
  width: ${({ theme, full, size, variant }) => {
    if (full) return '100%';
    if (variant === 'icon') {
      switch (size) {
        case 'xs':
          return theme.spaces[5];
        case 'sm':
          return theme.spaces[6];
        case 'md':
          return theme.spaces[7];
        case 'lg':
          return theme.spaces[8];
      }
    }
    return 'auto';
  }};
  height: ${({ theme, size, variant }) => {
    switch (size) {
      case 'xs':
        return theme.spaces[5];
      case 'sm':
        return theme.spaces[6];
      case 'md':
        return theme.spaces[7];
      case 'lg':
        return theme.spaces[8];
      default:
        if (size === '100%') {
          return size;
        }
        if (variant === 'icon' || variant === 'action') {
          return 'auto';
        }
        return theme.spaces[7];
    }
  }};
  padding: ${({ theme, variant, size, noPadding }) => {
    if (variant === 'icon' || variant === 'action' || noPadding) {
      return undefined;
    }
    switch (size) {
      case 'xs':
        return `${theme.spaces[1]} ${theme.spaces[2]}`;
      default:
        return `${theme.spaces[1]} ${theme.spaces[3]}`;
    }
  }};
`;

const ButtonStylesStyle = css<HubButtonProps>`
  ${({
    theme,
    variant,
    color,
    colorLevel,
    bgColor,
    bgColorLevel,
    borderColor,
    borderColorLevel
  }) => {
    const hash =
      color && colorLevel
        ? theme.colors[color][colorLevel]
        : color && theme.colors[color]
        ? theme.colors[color][500]
        : color
        ? color
        : theme.colors.primary[500];
    const bgHash =
      bgColor && bgColorLevel
        ? theme.colors[bgColor][bgColorLevel]
        : bgColor && theme.colors[bgColor]
        ? theme.colors[bgColor][500]
        : bgColor
        ? bgColor
        : 'transparent';
    const borderHash =
      borderColor && borderColorLevel
        ? theme.colors[borderColor][borderColorLevel]
        : borderColor && theme.colors[borderColor]
        ? theme.colors[borderColor][500]
        : borderColor
        ? borderColor
        : hash;

    switch (variant) {
      case 'outline':
        return `
          background-color: ${bgHash};
          color: ${hash};
          border: 1.5px solid ${borderHash};
        `;
      case 'ghost':
      case 'icon':
      case 'action':
        return `
          background-color: ${bgHash};
          color: ${hash};
        `;
      default:
        return `
          background-color: ${hash};
          color: ${theme.colors.white[500]};
        `;
    }
  }}
`;

const ButtonHoverStyle = css<HubButtonProps>`
  &:hover:not(:disabled) {
    ${({ theme, variant, color, bgHover }) => {
      const hash =
        color === 'white'
          ? theme.colors.neutralGrey[300]
          : color && theme.colors[color]
          ? shadeColor(theme.colors[color][500], -5)
          : color
          ? color
          : theme.colors.neutral[600];

      if (bgHover) {
        switch (variant) {
          case 'default':
            return `
              background-color: ${hash};
            `;
          default:
            return `
              background-color: ${theme.colors.neutralGrey[100]};
            `;
        }
      }

      switch (variant) {
        case 'outline':
          return `
            background-color: ${theme.colors.neutralGrey[100]};
          `;
        case 'ghost':
        case 'icon':
        case 'action':
          return `
            color: ${hash};
          `;
        default:
          return `
            background-color: ${hash};
          `;
      }
    }}
  }
`;

const ButtonActiveStyle = css<HubButtonProps>`
  &:not(:disabled) {
    ${({ theme, color, colorLevel }) => {
      const hash =
        color && colorLevel
          ? theme.colors[color][colorLevel]
          : color && theme.colors[color]
          ? theme.colors[color][500]
          : color
          ? color
          : theme.colors.primary[500];
      return `
        outline: 1px solid ${hash};
      `;
    }}
  }
`;

const ButtonDisabledStyle = css<HubButtonProps>`
  &:disabled {
    ${({ theme, variant }) => {
      switch (variant) {
        case 'outline':
          return `
            border-color: ${theme.colors.primary[100]};
            color: ${theme.colors.primary[100]};
          `;
        case 'ghost':
        case 'icon':
        case 'action':
          return `
            color: ${theme.colors.primary[100]};
          `;
        default:
          return `
            background-color: ${theme.colors.neutral[300]};
          `;
      }
    }}
  }
`;

const ButtonCommonStyle = css<HubButtonProps>`
  display: inline-flex;
  align-items: center;
  justify-content: center;
  gap: ${({ theme }) => theme.spaces[2]};
  border-radius: ${({ theme, rounded, radius }) =>
    rounded ? '50%' : theme.spaces[radius || 1]};
  border: none;
  font-weight: 600;
  transition: background-color, outline 0.5s;
  cursor: ${({ cursor, dragTrigger }) =>
    dragTrigger ? 'grab' : cursor || 'pointer'};
  white-space: nowrap;
  text-decoration: none;
`;

export const ButtonLoadingStyle = styled.div`
  width: ${({ theme }) => theme.spaces[5]};
  height: ${({ theme }) => theme.spaces[5]};
  border-radius: 50%;
  animation: 1s linear infinite ${() => animations.spin},
    ${() => animations.fadein} 1s ease;
`;

const ButtonShared = css<HubButtonProps>`
  ${ButtonCommonStyle}
  ${ButtonSizesStyle}
  ${ButtonStylesStyle}
  ${ButtonDisabledStyle}
  ${({ noHover }) => (noHover ? '' : ButtonHoverStyle)}
  ${({ active }) => (active ? ButtonActiveStyle : '')}

  ${({ clickable, disabled }) => {
    if (clickable === false) {
      return `
        pointer-events: none !important;
        cursor: not-allowed !important;
      `;
    }
    if (disabled) {
      return `
        cursor: not-allowed !important;
      `;
    }
    return '';
  }}

  ${ButtonLoadingStyle} {
    ${({ theme, variant, color, colorLevel }) => {
      switch (variant) {
        case 'outline':
        case 'ghost':
        case 'icon':
        case 'action':
          return `
              border: 3px solid ${
                color && theme.colors[color]
                  ? theme.colors[color][100]
                  : color
                  ? color
                  : theme.colors.primary[100]
              };
              border-top: 3px solid ${
                color && colorLevel
                  ? theme.colors[color][colorLevel]
                  : color && theme.colors[color]
                  ? theme.colors[color][500]
                  : color
                  ? color
                  : theme.colors.primary[300]
              };
            `;
        default:
          return `
              border: 3px solid rgba(255, 255, 255, 0.5);
              border-top: 3px solid ${
                color && colorLevel
                  ? theme.colors[color][colorLevel]
                  : color && theme.colors[color]
                  ? theme.colors[color][500]
                  : color
                  ? color
                  : theme.colors.white[500]
              };
            `;
      }
    }}
  }
`;

export const ButtonStyle = styled.button<HubButtonProps>`
  ${ButtonShared}
`;

export const ButtonLinkStyle = styled.a<HubButtonProps>`
  ${ButtonShared}
`;
