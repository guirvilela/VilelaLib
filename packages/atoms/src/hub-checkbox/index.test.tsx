import { render } from '@testing-library/react';
import React from 'react';

import { wrapse } from '@hubert/shared';
import { ThemeProvider } from '@hubert/styles';

import { HubCheckbox } from '.';

describe('HubCheckbox component', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <HubCheckbox id="test" checked>
        Test
      </HubCheckbox>,
      {
        wrapper: wrapse(ThemeProvider)
      }
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
