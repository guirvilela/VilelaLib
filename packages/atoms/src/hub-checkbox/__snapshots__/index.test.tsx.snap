// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HubCheckbox component matches snapshot 1`] = `
<DocumentFragment>
  .c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: background-color 0.5s;
  transition: background-color 0.5s;
}

.c3 {
  display: none;
}

.c4 {
  font-family: Nunito Sans;
  font-size: 14px;
  line-height: 1.6666666666666667;
  -webkit-letter-spacing: 0.01em;
  -moz-letter-spacing: 0.01em;
  -ms-letter-spacing: 0.01em;
  letter-spacing: 0.01em;
  font-weight: 400;
  margin-left: 8px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: start;
  -ms-flex-pack: start;
  justify-content: start;
}

.c0 .c1 {
  background-color: #083350;
  color: #FFFFFF;
  border: 1px solid transparent;
  border-radius: 4px;
  min-width: 16px;
  width: 16px;
  height: 16px;
}

<div
    class="c0"
    id="test"
  >
    <div
      class="c1 c2"
      id="hub-checkbox-icon-test"
    >
      <svg
        fill="none"
        height="8"
        viewBox="0 0 14 12"
        width="10"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M13.7 1.20039C13.3 0.800391 12.7 0.800391 12.3 1.20039L4.8 8.70039L1.7 5.60039C1.3 5.20039 0.7 5.20039 0.3 5.60039C-0.1 6.00039 -0.1 6.60039 0.3 7.00039L4.1 10.8004C4.3 11.0004 4.5 11.1004 4.8 11.1004C5.1 11.1004 5.3 11.0004 5.5 10.8004L13.7 2.60039C14.1 2.20039 14.1 1.60039 13.7 1.20039Z"
          fill="currentColor"
        />
      </svg>
    </div>
    <input
      checked=""
      class="c3"
      id="hub-checkbox-test"
      type="checkbox"
    />
    <label
      class="c4"
      for="hub-checkbox-test"
    >
      Test
    </label>
  </div>
</DocumentFragment>
`;
